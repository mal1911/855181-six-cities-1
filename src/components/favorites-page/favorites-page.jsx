import React, {PureComponent} from "react";
import Header from "../header/";
import PropTypes from "prop-types";
import {NavLink} from "react-router-dom";
import {connect} from "react-redux";
import FavoritesWrpapperData from "../favorites-wrapper-data";
import FavoritesWrpapperEmpty from "../favorites-wrapper-empty";
import withBodyClass from "../../hocs/with-body-class/with-body-class";
import {getCountResultFavorites, getLoadStatus, getError} from "../../reducer/favorites-data/selectors";
import {Operation} from "../../reducer/favorites-data/favorites-data";


class FavoritesPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.dispatch(Operation.loadData());
  }

  componentWillUnmount() {
  }

  render() {
    const FavoritesWrpapperEmptyBody = withBodyClass(FavoritesWrpapperEmpty, [`page--favorites-empty`]);
    return <React.Fragment>
      <Header/>
      {/*
        <FavoritesWrpapperEmptyBody/>
      */}
      <FavoritesWrpapperData/>
      <footer className="footer container">
        <NavLink className="footer__logo-link" to="/">
          <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33"/>
        </NavLink>
      </footer>
    </React.Fragment>;
  }
}

FavoritesPage.propTypes = {
  countResultFavorites: PropTypes.number.isRequired,
  loadStatus: PropTypes.bool,
  error: PropTypes.object,
  dispatch: PropTypes.func,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  countResultFavorites: getCountResultFavorites(state),
  loadStatus: getLoadStatus(state),
  error: getError(state),
});

export {FavoritesPage};

export default connect(mapStateToProps)(FavoritesPage);
