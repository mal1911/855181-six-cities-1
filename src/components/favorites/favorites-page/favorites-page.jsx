import React, {PureComponent} from "react";
import Header from "../../header/index";
import Footer from "../../footer/index";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import FavoritesWrpapperData from "../favorites-wrapper-data/index";
import FavoritesWrpapperEmpty from "../favorites-wrapper-empty/index";
import withBodyClass from "../../../hocs/with-body-class/with-body-class";
import withDataStatusScreen from "../../../hocs/with-data-status-screen/with-data-status-screen";
import {getCountResultFavorites, getFavoritesLoadStatus, getFavoritesError} from "../../../reducer/favorites-data/selectors";
import {Operation} from "../../../reducer/favorites-data/favorites-data";

class FavoritesPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.onDidMountComponent();
  }

  render() {
    const FavoritesWrpapperEmptyBody = withBodyClass(FavoritesWrpapperEmpty, [`page--favorites-empty`]);
    const WithDataStatusScreen = withDataStatusScreen(FavoritesWrpapperData, FavoritesWrpapperEmptyBody);
    return <React.Fragment>
      <Header/>
      <WithDataStatusScreen {...this.props}/>
      <Footer/>
    </React.Fragment>;
  }
}

FavoritesPage.propTypes = {
  countData: PropTypes.number.isRequired,
  loadStatus: PropTypes.bool,
  error: PropTypes.object,
  onDidMountComponent: PropTypes.func,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  countData: getCountResultFavorites(state),
  loadStatus: getFavoritesLoadStatus(state),
  error: getFavoritesError(state),
});

const mapDispatchToProps = (dispatch) => ({
  onDidMountComponent: () => {
    dispatch(Operation.loadFavoritesData());
  },
});

export {FavoritesPage};

export default connect(mapStateToProps, mapDispatchToProps)(FavoritesPage);
