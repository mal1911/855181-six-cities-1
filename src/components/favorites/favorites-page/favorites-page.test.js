import React from 'react';
import renderer from 'react-test-renderer';
import {FavoritesPage} from './favorites-page';
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../../reducer/name-spaces";

describe(`FavoritesPage correctly renders`, () => {
  const initialState = {};

  initialState[NameSpace.USER] = {
    isAuthorizationRequired: true,
    userObj: null,
    isUserLoading: true,
    userError: null,
  };

  initialState[NameSpace.DATA] = {
    activeCityIndex: 0,
    activeOrderIndex: 0,
    activeOfferId: 0,
    offersData: [],
    favoritesData: [],
    commentsData: [],
    offersError: null,
    favoritesError: null,
    commentsError: null,
    isLoading: true,
  };

  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(initialState);
    tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <FavoritesPage
              countData={1}
              loadStatus={false}
              error={null}
              onLoaded={jest.fn()}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
  });

  it(`FavoritesPage correctly renders`, () => {
    expect(tree).toMatchSnapshot();
  });
});
