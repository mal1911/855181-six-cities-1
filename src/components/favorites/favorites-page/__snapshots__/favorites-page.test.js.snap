// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage correctly renders FavoritesPage correctly renders 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            aria-current="page"
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                aria-current={null}
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "null",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        />
        ;
      </section>
    </div>
  </main>,
  <footer
    className="footer container"
  >
    <a
      aria-current="page"
      className="footer__logo-link active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="/img/logo.svg"
        width="64"
      />
    </a>
  </footer>,
]
`;
